File: backend-java/src/main/java/com/backend/java/backendjava/dtos/responses/CardResponse.java
package com.backend.java.backendjava.dtos.responses;

import com.backend.java.backendjava.infra.entities.Card;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CardResponse {
    private Long id;
    private String userDocument;
    private String creditCardToken;
    private Long cardValue;

    public CardResponse(Card card) {
        this.id = card.getId();
        this.userDocument = card.getUserDocument();
        this.creditCardToken = card.getCreditCardToken();
        this.cardValue = card.getCardValue();
    }
}

File: backend-kotlin/src/main/kotlin/com/backend/kotlin/backendkotlin/dtos/responses/CardResponse.kt
package com.backend.kotlin.backendkotlin.dtos.responses

import com.backend.kotlin.backendkotlin.infra.entities.Card

data class CardResponse(
    var id: Long?,
    var userDocument: String,
    var creditCardToken: String,
    var cardValue: Long
) {
    constructor(card: Card) : this(card.id, card.userDocument, card.creditCardToken, card.cardValue)
}
